src/Controls/src/Core/Routing.cs

```diff
@@ -143,7 +143,16 @@ public static Element GetOrCreateContent(string route)
 				// okay maybe its a type, we'll try that just to be nice to the user
 				var type = Type.GetType(route);
 				if (type != null)
-					result = Activator.CreateInstance(type) as Element;
+				{
+					if (Application.Current?.Handler?.MauiContext?.Services != null)
```

Comment:
Same comment here.  We'll probably need to add a parameter to this method that takes in an IMauiContext and propagate that into this call from the caller

Reply:
@PureWeen for global routes, do we need to pass in a context via GoToAsync() then, from where that's being called?

Reply:
Yes! you should be able to get the services via
_shell.FindMauiContext().Services